tools/install-powershell.ps1

```diff
@@ -0,0 +1,114 @@
+    }
+    Remove-Item -Path $Destination -Recurse -Force
+}
+New-Item -ItemType Directory -Path $Destination -Force > $null
+$Destination = Resolve-Path -Path $Destination | ForEach-Object -MemberName Path
+Write-Verbose "Destination: $Destination" -Verbose
+
+$architecture = Get-CimInstance win32_operatingsystem | ForEach-Object {
+                    switch ($_.OSArchitecture) {
+                        "64-bit" { "x64" }
+                        "32-bit" { "x86" }
+                        default  { throw "Daily package for OS architecture '$_' is not supported." }
+                    }
+                }
+$tempDir = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
+New-Item -ItemType Directory -Path $tempDir -Force > $null
+try {
+    if ($Daily) {
+        if (-not (Get-Module -Name PackageManagement -ListAvailable)) {
+            throw "PackageManagement module is required to install daily PowerShell Core."
+        }
+
+        if ($architecture -ne "x64") {
+            throw "The OS architecture is '$architecture'. However, we currently only support daily package for x64 Windows."
+        }
+
+        ## Register source if not yet
+        if (-not (Get-PackageSource -Name powershell-core-daily -ErrorAction SilentlyContinue)) {
+            $packageSource = "https://powershell.myget.org/F/powershell-core-daily"
+            Write-Verbose "Register powershell-core-daily package source '$packageSource' with PackageManagement" -Verbose
+            Register-PackageSource -Name powershell-core-daily -Location $packageSource -ProviderName nuget -Trusted > $null
+        }
+
+        $packageName = "powershell-win-x64-win7-x64"
+        $package = Find-Package -Source powershell-core-daily -AllowPrereleaseVersions -Name $packageName
+        Write-Verbose "Daily package found. Name: $packageName; Version: $($package.Version)" -Verbose
+        Install-Package -InputObject $package -Destination $tempDir -ExcludeVersion > $null
+
+        $contentPath = [System.IO.Path]::Combine($tempDir, $packageName, "content")
+        Copy-Item -Path $contentPath\* -Destination $Destination -Recurse -Force
+    } else {
+        $metadata = Invoke-RestMethod https://api.github.com/repos/powershell/powershell/releases/latest
+        $release = $metadata.tag_name -replace '^v'
+
+        $packageName = "PowerShell-${release}-win-${architecture}.zip"
+        $downloadURL = "https://github.com/PowerShell/PowerShell/releases/download/v${release}/${packageName}"
+        Write-Verbose "About to download package from '$downloadURL'" -Verbose
+
+        $packagePath = Join-Path -Path $tempDir -ChildPath $packageName
+        Invoke-WebRequest -Uri $downloadURL -OutFile $packagePath
+
+        Expand-Archive -Path $packagePath -DestinationPath $Destination
+    }
+
+    if ($AddToPath -and (-not $env:Path.Contains($Destination))) {
```

Comment:
Is there a case where the user wouldn't addtopath?  Seems like this would be the primary use case?  Perhaps have `-DoNotAddToPath` instead?

Reply:
I think when installing daily build package, a user probably don't want it to be added to Path by default.

Reply:
I'm thinking the opposite where for daily self-hosting, I would want it in the path, although it would only need to be added once, so this isn't a big deal.  I'm fine leaving it the way it is.

Reply:
OK. I will leave it unchanged.